### PR TITLE
refactor: simplify configurations

### PR DESCRIPTION
**Pull request by JounQin**
I want to work on #207, and found the typescript build steps are a bit redundant. There is no need to declare `tsc -b` in all packages, we've already had `references`: https://github.com/markuplint/markuplint/pull/210/files#diff-b55cdbef4907b7045f32cc5360d48d262cca5f94062e353089f189f4460039e0R6-R76

TypeScript itself will compile all related modules automatically.

For `devDependencies`, I move all of them into the root `package.json` so that no duplicate packages will be installed in different `node_modules`. `webpack/yarn.lock` can be deleted.

https://github.com/markuplint/markuplint/pull/210/files#diff-a4d4f2e0939888fa32e5e7eb7a17fc2a84aaabe8d5c867443da4351751158313

`lerna bootstrap` is not needed because of `yarn workspace`.

For ESLint, `@typescript-eslint` should only be enabled for `*.ts,*.tsx` files. Then a lot of ESLint disable comments for `@typescript-eslint/*` can be deleted.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,28 +1,27 @@
...
 		"test": "jest --coverage",
 		"test:coverage": "jest --coverage --coverageReporters=text-lcov | coveralls",
 		"test:clear": "jest --clearCache",
 		"test:unit": "jest",
 		"test:heapdump": "node --expose-gc ./heaptest/index.js",
-		"lint": "prettier --write './{packages,website,playground}/**/*.{ts,tsx,json}'; eslint --fix './{packages,website,playground}/**/*.{ts,tsx}'; yarn cli './website/**/*.tsx'",
-		"schema": "lerna run schema",
-		"gen": "cd packages/\\@markuplint/html-spec/; yarn gen; cd ../",
-		"all": "yarn lint; yarn clean; yarn gen; yarn build; yarn schema; yarn test:clear; yarn test;",
+		"lint": "prettier --write .; eslint . --fix; yarn cli './website/**/*.tsx'",
```
Comment (JounQin) [inline package.json:12]: All source codes except test fixtures should be formatted and linted.

`.prettierignore` and `.eslintignore` makes sure IDE works correctly too.

**Comment by JounQin** [comment]
Besides, a lot of packages are outdated.

```log
yarn outdated v1.22.10
info Color legend : 
 "<red>"    : Major Update backward-incompatible updates 
 "<yellow>" : Minor Update backward-compatible features 
 "<green>"  : Patch Update backward-compatible bug fixes
Package                          Current    Wanted      Latest      Workspace                 Package Type    URL                                                                                 
@textlint/kernel                 3.4.5      3.4.5       12.0.0      @markuplint/rule-textlint dependencies    https://github.com/textlint/textlint/tree/master/packages/@textlint/textlint-kernel/
@types/cheerio                   0.22.28    0.22.29     0.22.29                               devDependencies https://github.com/DefinitelyTyped/DefinitelyTyped.git                              
@types/meow                      5.0.0      5.0.0       6.0.0                                 devDependencies https://github.com/DefinitelyTyped/DefinitelyTyped.git                              
@types/mini-css-extract-plugin   0.9.1      0.9.1       1.4.3                                 devDependencies https://github.com/DefinitelyTyped/DefinitelyTyped.git                              
@types/node                      14.14.44   14.17.4     15.12.5                               devDependencies https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/node           
@typescript-eslint/eslint-plugin 4.23.0     4.28.0      4.28.0                                devDependencies https://github.com/typescript-eslint/typescript-eslint#readme                       
autoprefixer                     9.8.6      9.8.6       10.2.6                                devDependencies https://github.com/postcss/autoprefixer#readme                                      
cheerio                          1.0.0-rc.9 1.0.0-rc.10 1.0.0-rc.10                           devDependencies https://cheerio.js.org/                                                             
css-loader                       4.3.0      4.3.0       5.2.6                                 devDependencies https://github.com/webpack-contrib/css-loader                                       
cssnano                          4.1.11     4.1.11      5.0.6                                 devDependencies https://github.com/cssnano/cssnano                                                  
eslint                           7.26.0     7.29.0      7.29.0                                devDependencies https://eslint.org                                                                  
eslint-config-next               11.0.0     11.0.1      11.0.1                                devDependencies https://github.com/vercel/next.js#readme                                            
eslint-plugin-jsdoc              31.6.1     31.6.1      35.4.0                                devDependencies https://github.com/gajus/eslint-plugin-jsdoc#readme                                 
execa                            5.0.0      5.1.1       5.1.1                                 devDependencies https://github.com/sindresorhus/execa#readme                                        
get-stdin                        8.0.0      8.0.0       9.0.0       markuplint                dependencies    https://github.com/sindresorhus/get-stdin#readme                                    
html-loader                      1.3.2      1.3.2       2.1.2                                 devDependencies https://github.com/webpack-contrib/html-loader                                      
html-webpack-plugin              4.5.2      4.5.2       5.3.2                                 devDependencies https://github.com/jantimon/html-webpack-plugin                                     
meow                             9.0.0      9.0.0       10.0.1      markuplint                dependencies    https://github.com/sindresorhus/meow#readme                                         
mini-css-extract-plugin          0.10.1     0.10.1      1.6.1                                 devDependencies https://github.com/webpack-contrib/mini-css-extract-plugin                          
monaco-editor                    0.20.0     0.20.0      0.25.2      @markuplint/playground    dependencies    https://github.com/Microsoft/monaco-editor#readme                                   
monaco-editor-webpack-plugin     1.9.1      1.9.1       4.0.0                                 devDependencies https://github.com/Microsoft/monaco-editor-webpack-plugin#readme                    
next                             11.0.0     11.0.0      11.0.1      @markuplint/website       dependencies    https://nextjs.org                                                                  
postcss-calc                     7.0.5      7.0.5       8.0.0                                 devDependencies https://github.com/postcss/postcss-calc#readme                                      
postcss-custom-media             7.0.8      7.0.8       8.0.0                                 devDependencies https://github.com/postcss/postcss-custom-media#readme                              
postcss-loader                   3.0.0      3.0.0       6.1.0                                 devDependencies https://github.com/webpack-contrib/postcss-loader                                   
prettier                         2.3.0      2.3.2       2.3.2                                 devDependencies https://prettier.io                                                                 
sass-loader                      9.0.3      9.0.3       12.1.0                                devDependencies https://github.com/webpack-contrib/sass-loader                                      
strip-ansi                       6.0.0      6.0.0       7.0.0       markuplint                dependencies    https://github.com/chalk/strip-ansi#readme                                          
style-loader                     1.3.0      1.3.0       3.0.0                                 devDependencies https://github.com/webpack-contrib/style-loader                                     
textlint                         11.9.1     11.9.1      12.0.0      @markuplint/rule-textlint dependencies    https://github.com/textlint/textlint/                                               
ts-loader                        8.2.0      8.3.0       9.2.3                                 devDependencies https://github.com/TypeStrong/ts-loader                                             
typedoc                          0.19.2     0.19.2      0.21.2                                devDependencies https://typedoc.org                                                                 
webpack                          4.46.0     4.46.0      5.40.0                                devDependencies https://github.com/webpack/webpack                                                  
webpack-cli                      3.3.12     3.3.12      4.7.2                                 devDependencies https://github.com/webpack/webpack-cli/tree/master/packages/webpack-cli
```

File: packages/@markuplint/ejs-parser/package.json [hunk 2]
```diff
@@ -11,13 +11,10 @@
 	"publishConfig": {
 		"access": "public"
 	},
-	"scripts": {
-		"build": "tsc",
```
Comment (JounQin) [inline packages/@markuplint/ejs-parser/package.json:15]: All `tsc` usage in sub packages are removed.

File: tsconfig.json [hunk 3]
```diff
@@ -1,22 +1,77 @@
...
-		"skipLibCheck": true,
-		"baseUrl": ".",
-		"paths": {
-			"@markuplint/*": ["./packages/@markuplint/*/src"],
-			"*": ["./packages/*/src"]
-		}
+		"noEmit": true
 	},
-	"exclude": ["node_modules"]
+	"references": [
```
Comment (JounQin) [inline tsconfig.json:6]: Sort here matters.

File: playground/tsconfig.json [hunk 4]
```diff
@@ -1,9 +1,12 @@
 {
-	"extends": "../tsconfig.json",
+	"extends": "../tsconfig.base",
 	"compilerOptions": {
-		"resolveJsonModule": true,
 		"declaration": false,
-		"importHelpers": true
+		"module": "ESNext"
```
Comment (JounQin) [inline playground/tsconfig.json:5]: Not sure why, it's required to get working for `svelte-loader`.

File: playground/tsconfig.json [hunk 5]
```diff
@@ -1,9 +1,12 @@
...
 	},
-	"exclude": ["node_modules"]
+	"ts-node": {
+		"compilerOptions": {
+			"module": "CommonJS"
```
Comment (JounQin) [inline playground/tsconfig.json:9]: For `webpack.config.ts`

File: website/.eslintrc [hunk 6]
```diff
@@ -1,5 +1,5 @@
 {
-	"extends": ["../.eslintrc", "plugin:react/recommended", "plugin:react-hooks/recommended"],
```
Comment (JounQin) [inline website/.eslintrc:2]: Extending from parent directory is not needed.

**Comment by YusukeHirao** [comment]
@JounQin 

Thanks! But your PR is huge as usual...
Could you split this PR by some purpose?
For example:
- Tidying dependencies.
- Fixing typos and renaming.
- Tidying and improving scripts and commands.
- Updating versions of dependent packages.

And, I confuse if move devDependencies to the root package.json. Because I become to never know what sub-packages depends. I might move some packages to another repository. I want to keep clear dependencies.

**Comment by JounQin** [comment]
We can start from #209

**Comment by YusukeHirao** [comment]
@JounQin 

I merged #209.

**Comment by YusukeHirao** [comment]
@JounQin 

I started to improve for [v2](https://github.com/markuplint/markuplint/tree/v2).

**Pull request closed without merging**